### ISSUE TITLE
[API Proposal]: Quaternion.Zero is missing

### ISSUE THREAD

**Issue by michael-hawker** [dotnet/runtime#57253]
### Background and motivation

Vector2, Vector3, and Vector4 when created with new() are all zeroed out and have corresponding `.Zero` static properties. `Quaternion` on the otherhand only has the `Identity` static property and doesn't have the corresponding `Zero` one to represent its base state when created with `new()`.

### API Proposal

```C#
namespace System.Numerics
{
     public class Quaternion
     {
        /// <summary>Gets a newly initialized quaternion.</summary>
        /// <value>A quaternion whose values are <c>(0, 0, 0, 0)</c>.</value>
        public static Quaternion Zero
        {
            get => new();
        }
     }
}
```     

### API Usage

```C#
// New Quaternion
var q = Quaternion.Zero;
q.X = 0.5;

// Checking
if (q == Quaternion.Zero)
{
  // Was uninitialized
}
```     

### Risks

No breaking changes, adds value to matching base struct types in the System.Numerics package for base initialization state.

**Comment by GrabYourPitchforks** [comment]
`Quaternion` also has an _IsIdentity_ property. If this API is approved, would _IsZero_ come along for the ride for symmetry?

**Comment by michael-hawker** [comment]
> `Quaternion` also has an _IsIdentity_ property. If this API is approved, would _IsZero_ come along for the ride for symmetry?

That's a great question, would it make sense to have `IsZero` and `IsOne` type function on `Vector2`, `Vector3`, and `Vector4` too for symmetry with the `Quaternion` function too?

**Comment by hrrrrustic** [comment]
@GrabYourPitchforks @tannergooding Is there any reason why `Quaternion` fields are mutable?
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Numerics/Quaternion.cs#L14-L27

It's easy to make whole struct readonly, just need to rewrite code like that
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Numerics/Quaternion.cs#L76-L81

**Comment by GrabYourPitchforks** [comment]
@hrrrrustic I suspect it's mostly that there wasn't a whole lot of consistency when these types were first introduced all those years ago. Maybe partly that the original designers believed that updating individual values would be a common operation and that apps didn't want to incur the cost of updating the entire struct? (Keep in mind - this was originally designed for a much older JIT / platform.)

**Comment by hrrrrustic** [comment]
Well, I'd like to propose to make the whole struct readonly at least for consistency with other numbers/numerics-like types
But I'll wait for @tannergooding answer, maybe he has some strong arguments against that

**Comment by svick** [comment]
@hrrrrustic That would be a breaking change, so I don't see that happening.

**Comment by tannergooding** [comment]
Yep, its a breaking change since the fields are `public` and so we cannot make the fix.

Ideally these would be `readonly`, but since they can't be all the methods should (and I believe already have been updated to be) readonly instead.

**Comment by tannergooding** [comment]
I think exposing `Zero` makes sense here. I'm not convinced `IsZero` is worthwhile to expose, developers can simply use `value == Quaternion.Zero`, which will be more efficient.

**Comment by bartonjs** [comment]
Looks good as proposed.

```C#
namespace System.Numerics
{
     partial struct Quaternion
     {
        /// <summary>Gets a newly initialized quaternion.</summary>
        /// <value>A quaternion whose values are <c>(0, 0, 0, 0)</c>.</value>
        public static Quaternion Zero
        {
            get => new();
        }
     }
}
```

**Comment by tannergooding** [comment]
Marked this as `up-for-grabs`, but note that properly implementing this may require JIT side work to ensure the codegen is optimal like for the `Vector4` scenario.
